src/faultlog/util.cpp

```diff
@@ -23,4 +32,190 @@ std::string getGuardReason(const GuardRecords& guardRecords,
     }
     return "Unknown";
 }
+ProgressStages getBootProgress(sdbusplus::bus::bus& bus)
+{
+    try
+    {
+        using PropertiesVariant =
+            sdbusplus::utility::dedup_variant_t<ProgressStages>;
+
+        auto retVal = readProperty<PropertiesVariant>(
+            bus, "xyz.openbmc_project.State.Host",
+            "/xyz/openbmc_project/state/host0",
+            "xyz.openbmc_project.State.Boot.Progress", "BootProgress");
+        const ProgressStages* progPtr = std::get_if<ProgressStages>(&retVal);
+        if (progPtr != nullptr)
+        {
+            return *progPtr;
+        }
+    }
+    catch (const std::exception& ex)
+    {
+        lg2::error("Failed to read Boot Progress property {ERROR}", "ERROR",
+                   ex.what());
+    }
+
+    lg2::error("Failed to read Boot Progress state value");
+    return ProgressStages::Unspecified;
+}
+
+HostState getHostState(sdbusplus::bus::bus& bus)
+{
+    try
+    {
+        using PropertiesVariant =
+            sdbusplus::utility::dedup_variant_t<HostState>;
+
+        auto retVal = readProperty<PropertiesVariant>(
+            bus, "xyz.openbmc_project.State.Host",
+            "/xyz/openbmc_project/state/host0",
+            "xyz.openbmc_project.State.Host", "CurrentHostState");
+        const HostState* statePtr = std::get_if<HostState>(&retVal);
+        if (statePtr != nullptr)
+        {
+            return *statePtr;
+        }
+    }
+    catch (const std::exception& ex)
+    {
+        lg2::error("Failed to read host state property {ERROR}", "ERROR",
+                   ex.what());
+    }
+
+    lg2::error("Failed to read host state value");
+    return HostState::Off;
+}
+
+bool isHostAppliedGuardRecords(sdbusplus::bus::bus& bus)
+{
+    ProgressStages progress = getBootProgress(bus);
+    if ((progress == ProgressStages::MemoryInit) ||
+        (progress == ProgressStages::SecondaryProcInit) ||
+        (progress == ProgressStages::PCIInit) ||
+        (progress == ProgressStages::SystemInitComplete) ||
+        (progress == ProgressStages::SystemSetup) ||
+        (progress == ProgressStages::OSStart) ||
+        (progress == ProgressStages::OSRunning))
+    {
+        return true;
+    }
+    return false;
+}
+bool isHostProgressStateRunning(sdbusplus::bus::bus& bus)
+{
+    ProgressStages progress = getBootProgress(bus);
+    if ((progress == ProgressStages::SystemInitComplete) ||
+        (progress == ProgressStages::SystemSetup) ||
+        (progress == ProgressStages::OSStart) ||
+        (progress == ProgressStages::OSRunning))
+    {
+        return true;
+    }
+    return false;
+}
+
+bool isHostStateRunning(sdbusplus::bus::bus& bus)
+{
+    HostState hostState = getHostState(bus);
+    if (hostState == HostState::Running)
```

Comment:
`return (hostState == HostState::Running);` would club the next two lines into one. Or more better,
`return getHostState(bus) == HostState::Running;`. Same can be applied in the above function